### PR TITLE
[torch] Implement aten::native_batch_norm.out for CPU

### PR DESCRIPTION
**Pull request by larryliu0820**
Summary:
Implement `native_batch_norm.out` for CPU. Reuses the main logic for `native_batch_norm` but extract out the Tensor creation logic for outputs. There are 3 outputs: `output`, `save_mean` and `save_var`. `batch_norm_cpu` calls `batch_norm_cpu_update_stats_template` to get `save_mean` and `save_var`, and then calls into `batch_norm_cpu_transform_input_template` which initializes `output`.

In the implementation of `batch_norm_cpu_out`, I did the following:

* Let `batch_norm_cpu_transform_input_template` to take another argument `output`, ask the call sites to pass in a output Tensor.

* Overload `batch_norm_cpu_update_stats_template` to take `save_mean` and `save_var`, ask the call sites to pass in those Tensors.

* In `batch_norm_cpu_out`, pass `output`, `save_mean` and `save_var` all the way to our new `batch_norm_cpu_transform_input_template` and `batch_norm_cpu_update_stats_template`.

* In `batch_norm_cpu`, prepare for these outputs and call `batch_norm_cpu_out`.

Test Plan: Enable unit tests for `native_batch_norm.out`.

Differential Revision: D40992036

### CODE REVIEW

**Review by iseeyuan — APPROVED** [review]
Nice! Thanks @larryliu0820 !

File: aten/src/ATen/native/Normalization.cpp [hunk 1]
```diff
@@ -123,17 +123,16 @@ std::tuple<Tensor,Tensor,Tensor> batch_norm_cpu_transform_input_template(
     const Tensor& input, const Tensor& weight, const Tensor& bias,
     const Tensor& save_mean /* optional */, const Tensor& save_invstd /* optional */,
     const Tensor& running_mean /* optional */, const Tensor& running_var /* optional */,
-    bool train, double eps) {
+    bool train, double eps, Tensor& output) {
 
   bool all_contiguous = is_contiguous(input)
```
Comment (jjsjann123) [inline aten/src/ATen/native/Normalization.cpp:128]: shouldn't `all_contiguous` also checks contiguity on outputs then?

File: aten/src/ATen/native/Normalization.cpp [hunk 2]
```diff
@@ -690,40 +694,87 @@ std::tuple<Tensor, Tensor> batch_norm_update_stats_cpu(
...
+  Tensor output = at::empty_like(self, all_contiguous ? suggest_memory_format_contig(self) : self.suggest_memory_format());
+
+  // Prepare save_mean and save_var
+  Tensor save_var;
+  Tensor save_mean;
+  const bool mixed_type = is_mixed_type(self, weight, bias, running_mean, running_var);
+  const int64_t ndim = self.dim();
+  DimVector reduce_dims(ndim - 1);
+  reduce_dims[0] = 0;
+  for (const auto i : c10::irange(2, ndim)) {
```
Comment (jjsjann123) [inline aten/src/ATen/native/Normalization.cpp:757]: nitpick `std::iota` maybe?

**Review by jjsjann123 — APPROVED** [review]
LGTM, minor comments
